src/Development/IDE/Core/FileStore.hs

```diff
@@ -196,7 +203,20 @@ setFileModified state nfp = do
           liftIO $ progressUpdate KickStarted
           void $ use GetSpanInfo nfp
           liftIO $ progressUpdate KickCompleted
-    shakeRestart state [da]
+        parents = mkDelayedAction "ParentTC" L.Debug (typecheckParents nfp)
+    shakeRestart state $
+      [da] ++ [parents | prop]
```

Comment:
This call to `shakeRestart` requests the parents type checking as part of the system actions, which are never retried. Is that what you want? 
If you wanted to have them retried, you would need to use `shakeEnqueue` or something similar after calling `shakeRestart`.